First draft for user guide PR (#166)
neoforged/Documentation: userguide/docs/launchers.md

---
sidebar_position: 3
---

# Launchers and Mod Managers

> There are two supported ways that you can launch Minecraft with Neoforge.
> 
> The first is to use the official, vanilla launcher from Mojang, which the NeoForge installer creates a profile for.
> 
> The second option is to use a third-party launcher, some of which are listed below in the [Third-Party Launchers] section.

Reviewer: ```suggestion
There are two ways to launch Minecraft with Neoforge:
- The first option is to use the official Minecraft launcher provided by Mojang, which the NeoForge installer creates a profile in.
- The second option is to use a third-party launcher, which usually provide extra functionality and some of which are listed below in the [Third-Party Launchers] section.
```

## Locating the Game Directory

> Minecraft stores its files, like your game's saves, as well as crash reports and log files in a so-called game directory.
> 
> The location of the game directory will differ depending on which launcher you use to run the game.

Reviewer: ```suggestion
Minecraft stores its files, such as your game's save files, log files and crash reports, in a so-called game directory. The location of the game directory will differ depending on which launcher you use to run the game.
```

> The vanilla launcher will create a `.minecraft` directory in your OS user's home directory.  This is the game directory for the vanilla launcher.

Reviewer: Mention here that this can be changed in the profile options.

> <!-- List ot link out to where to find user home directory? -->

Reviewer: Would probably be beneficial, yes.

> :::note

Reviewer: No note block necessary imo.

If you are using another launcher, refer to their documentation on where the game directory is located.
:::

## Locating Log Files and Crash Reports

> Minecraft stores log files and crash reports in the game directory.

Reviewer: Merge with next paragraph.

You might be asked to provide log files or crash reports when seeking help.

If you do not know where to find them, open your game directory in a file manager application where there should be two directories:

> - `logs` This directory contains log files for the game, with the files `latest.log` and `debug.log` being most pertinent.

Reviewer: ```suggestion
- `logs`: This directory contains log files for the game, with the files `latest.log` and `debug.log` being most pertinent.
```

> - `crash-reports` This directory contains plain-text crash reports, each with timestamped filenames e.g. `crash-2024-08-11_15.54.06-client.txt`.

Reviewer: ```suggestion
- `crash-reports`: This directory contains plain-text crash reports, each with timestamped filenames e.g. `crash-2024-08-11_15.54.06-client.txt`. If a crash occurs too early, this directory might not exist; please provide the logs found in the `logs` directory (as outlined above) instead.
```

> <!-- Not sure if the above list is the best way to format the info in it... -->

Reviewer 2: I would put more stock in the debug log as the latest log usually doesn't contain enough info. I don't believe we usually ask for the crash report file since it's in the debug log, but it's good to know about.

> ## Third-Party Launchers

Reviewer 2: Ok, not sure how much I support this section as it seems like we're giving recommendations on third party launchers. I don't find any major issue with it, but I'm not as well versed in launchers to determine whether they are not doing weird things with the access tokens or not.

I will defer this to @XFactHD and @IchHabeHunger54.

Reviewer: Yeah, this was originally my idea, so I'll take care of this.

> There are a number of third-party launchers (also known as mod managers) to choose from, including:

Reviewer: I would restructure this subsection as a whole. In my opinion, this should be something along the lines of:

```
There are a number of third-party launchers developed for Minecraft that typically have additional features for modded users. These allow the users to manage different modpacks, install mods themselves etc. /* more info here maybe */

Popular third-party launchers include /* maybe list some pros for each; add links */:

- CurseForge
- Modrinth
- ATLauncher
- Prism Launcher
- FTB?
- GDLauncher
```

- [CurseForge App]
- [Modrinth App]
- [ATLauncher]
- [Prism Launcher]

These launchers differ from the vanilla launcher in a number of helpful ways.

First of all, the launcher should fully handle installing and updating NeoForge for you.

Second, you will be able to create multiple game profiles (also known as instances), allowing you to easily seperate different versions of the game.

Finally, all of the linked launchers have the ability to manage mods, making installing and updating mods much easier.

> [Third-Party Launchers]: ./launchers.md#third-party-launchers

Reviewer: ```suggestion
[Third-Party Launchers]: #third-party-launchers
```
Internal references like this are supported and the common style in the rest of the docs.
